### PR TITLE
LPD-30530 Add ERC to roles

### PR DESCRIPTION
**Pull request by smottal**
https://liferay.atlassian.net/browse/LPD-30530

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:report:48430498

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:sf

File: portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java [hunk 1]
```diff
@@ -1874,6 +1860,31 @@ public void unsetUserRoles(long userId, long[] roleIds)
 		reindex(userId);
 	}
 
+	@Override
+	public Role updateExternalReferenceCode(
+			long roleId, String externalReferenceCode)
```
Comment (andrea-ale-sbarra) [inline portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:1865]: I'm not sure ERC should be the second parameter

File: modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java [hunk 2]
```diff
@@ -110,6 +110,27 @@ public void testUpgradeProcess() throws Exception {
 			addExternalReferenceCodeModels(String tableName)
 		throws PortalException;
 
+	protected void assertExternalReferenceCode(
```
Comment (andrea-ale-sbarra) [inline modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java:113]: why did you change this class?

**Review by andrea-ale-sbarra — COMMENTED** [review]
Overall looks good, just a check to be sure before submitting.

Thank you!

File: modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java (hunk 2 above)
Comment (smottal) [reply modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java:113]: because I need to override some method because in the role table there is no the groupId column (that is checked in the base one)

File: portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java (hunk 1 above)
Comment (smottal) [reply portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:1865]: it's very "random"
```
public OAuth2Application updateExternalReferenceCode(
			long oAuth2ApplicationId, String externalReferenceCode)
```
but also
```
public CommercePriceEntry updateExternalReferenceCode(
			String externalReferenceCode, CommercePriceEntry commercePriceEntry)
```
but also
```
public CommerceTierPriceEntry updateExternalReferenceCode(
			CommerceTierPriceEntry commerceTierPriceEntry,
			String externalReferenceCode)
```

**Comment by smottal** [comment]
ci:forward:force
